lib/src/command_runner.dart

```diff
@@ -45,14 +46,28 @@ class PubCommandRunner extends CommandRunner {
   /// Returns an empty string if no command is being run. (This is only
   /// expected to happen when unit tests invoke code inside pub without going
   /// through a command.)
+  ///
+  /// For top-level commands, if an alias is used, the primary command name is
+  /// returned. For instance `install` becomes `get`.
   static String get command {
     if (_options == null) return '';
 
     var list = <String>[];
     for (var command = _options.command;
         command != null;
         command = command.command) {
-      list.add(command.name);
+      var commandName = command.name;
+
+      if (list.isEmpty) {
```

Comment:
Nit: This could have been before the for loop - that's when the list is empty.

Reply:
True, but I wanted to keep a single place where we add the command to the list. At least I felt that made the code easier to reason about.